### PR TITLE
Fetching data from API & SIWE

### PR DESCRIPTION
**Pull request by efstajas**

### CODE REVIEW

File: src/lib/stores/workstreams/types.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+export interface WorkstreamBase {
+    id: string;
+    state: WorkstreamState;
+    creator: string;
+    created_at: Timestamp;
+    payment_rate?: number;
+    payment_currency?: Currency;
+    title: string;
+    desc: string;
+}
```
Comment (juliendonck) [inline src/lib/stores/workstreams/types.ts:31]: Wouldn't these have "applications: Application" as well?

**Review by juliendonck — APPROVED** [review]
Looks good overall to me. Left a few inline comments.

File: src/lib/stores/workstreams/types.ts (hunk 1 above)
Comment (efstajas) [reply src/lib/stores/workstreams/types.ts:31]: yep, still need to add that 👍

File: src/lib/stores/workstreams/types.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export type Application = {
+    title: string;
+    proposal_id: number;
+    desc: string;
+    creator: string;
+    recipients: Array<string>;
+    payment_rate: number;
+    currency: string;
+    created_at: number;
+    ending_at: number;
```
[the remark this answers is not in the document]
Comment (efstajas) [reply src/lib/stores/workstreams/types.ts:53]: i just carried over that type from what was already there, gonna take a look at applications now 👍

**Pull request merged**